File: frontend/src/app/api/motivo-peso-manual.ts
// Catálogo de motivos de peso manual (decisión de producto #6 / diseño D2).
// Lista cerrada en código, sin pantalla admin y sin sync — cambiarla es un
// despliegue. Los códigos son los identificadores del enum central
// `MotivoPesoManual` (backend/Domain/Boletas/MotivoPesoManual.cs) y del const
// `MOTIVOS_PESO_MANUAL` de frontend/electron/db.ts, VERBATIM. El servidor local
// los envía en `GET /estado` (`motivosPesoManual`); acá viven las etiquetas en
// español que ve el operador y la consulta.

export type MotivoPesoManual =
  | 'IndicadorSinSenal'
  | 'IndicadorEnReparacion'
  | 'CorteEnergia'
  | 'Otro';

/** Único motivo que exige un detalle libre obligatorio. */
export const MOTIVO_PESO_MANUAL_OTRO: MotivoPesoManual = 'Otro';

export const LABELS_MOTIVO_PESO_MANUAL: Record<MotivoPesoManual, string> = {
  IndicadorSinSenal: 'Indicador sin señal',
  IndicadorEnReparacion: 'Indicador en reparación',
  CorteEnergia: 'Corte de energía',
  Otro: 'Otro',
};

/** Etiqueta en español de un código de motivo; devuelve el código crudo si no está mapeado. */
export function etiquetaMotivoPesoManual(codigo: string | null | undefined): string {
  if (codigo === null || codigo === undefined || codigo === '') return '—';
  return (LABELS_MOTIVO_PESO_MANUAL as Record<string, string>)[codigo] ?? codigo;
}

File: backend/Domain/Boletas/MotivoPesoManual.cs
namespace SmsBackend.Domain.Boletas;

/// <summary>
/// Catálogo fijo de motivos por los que un peso se tipeó manualmente en lugar
/// de leerlo del indicador. Es una lista cerrada en código (decisión de
/// producto #6): sin pantalla de administración, sin sincronización — cambiarla
/// es un despliegue. El renderer mapea el código a su etiqueta en español.
/// </summary>
public enum MotivoPesoManual
{
    /// <summary>"indicador sin señal".</summary>
    IndicadorSinSenal,

    /// <summary>"indicador en reparación".</summary>
    IndicadorEnReparacion,

    /// <summary>"corte de energía".</summary>
    CorteEnergia,

    /// <summary>"otro" — requiere el detalle de texto libre.</summary>
    Otro,
}

/// <summary>
/// Utilidades de parseo del catálogo <see cref="MotivoPesoManual"/>. El backend
/// mantiene su propio enum (no puede importar el <c>MOTIVOS_PESO_MANUAL</c> de
/// <c>db.ts</c>); la duplicación de los 4 valores entre runtimes está fijada por
/// los tests de ingesta de sync.
/// </summary>
public static class MotivosPesoManual
{
    /// <summary>
    /// Intenta resolver un código de catálogo. Rechaza null/blanco, cadenas
    /// numéricas y cualquier valor que no sea uno de los 4 nombres definidos.
    /// </summary>
    public static bool TryParse(string? valor, out MotivoPesoManual motivo)
    {
        motivo = default;
        if (string.IsNullOrWhiteSpace(valor) || valor.All(char.IsDigit))
        {
            return false;
        }

        return Enum.TryParse(valor, ignoreCase: true, out motivo) && Enum.IsDefined(motivo);
    }
}
